test/rules/best-practises/no-empty-blocks.js

```diff
@@ -1,27 +1,75 @@
-const { assertNoWarnings, assertErrorMessage, assertWarnsCount } = require('../../common/asserts')
+const {
+  assertNoWarnings,
+  assertErrorMessage,
+  assertWarnsCount,
+  assertErrorCount,
+} = require('../../common/asserts')
 const linter = require('../../../lib/index')
 const { contractWith, funcWith } = require('../../common/contract-builder')
 
 describe('Linter - no-empty-blocks', () => {
-  const EMPTY_BLOCKS = [
-    funcWith('if (a < b) {  }'),
-    contractWith('struct Abc {  }'),
-    contractWith('enum Abc {  }'),
-    contractWith('constructor () {  }'),
-    'contract A { }',
-    funcWith('assembly {  }'),
-  ]
+  describe('Basic empty blocks', () => {
+    const EMPTY_BLOCKS = [
+      funcWith('if (a < b) {  }'),
+      funcWith('for(uint i = 0; i < 10; i++) {  }'),
+      funcWith('while(true) {  }'),
+      funcWith('do {  } while(true);'),
+      funcWith('try foo() {  } catch {  }'),
+      contractWith('struct Abc {  }'),
+      contractWith('enum Abc {  }'),
+      contractWith('constructor () {  }'),
+      'contract A { }',
+      funcWith('assembly {  }'),
+    ]
 
-  EMPTY_BLOCKS.forEach((curData) =>
-    it(`should raise warn for empty blocks ${label(curData)}`, () => {
-      const report = linter.processStr(curData, {
-        rules: { 'no-empty-blocks': 'warn' },
+    EMPTY_BLOCKS.forEach((curData) =>
+      it(`should raise warn for empty block: ${label(curData)}`, () => {
+        const report = linter.processStr(curData, {
+          rules: { 'no-empty-blocks': 'warn' },
+        })
+        assertWarnsCount(report, curData.includes('try') ? 2 : 1)
+        assertErrorMessage(report, 'empty block')
       })
+    )
+  })
 
-      assertWarnsCount(report, 1)
-      assertErrorMessage(report, 'empty block')
+  describe('Valid non-empty blocks', () => {
+    const VALID_BLOCKS = [
+      funcWith('if (a < b) { uint x = 1; }'),
+      funcWith('for(uint i = 0; i < 10; i++) { sum += i; }'),
+      funcWith('while(true) { count++; }'),
+      funcWith('do { count--; } while(true)'),
+      funcWith('try foo() { emit Event(); } catch { revert(); }'),
+      contractWith('struct Point { uint x; uint y; }'),
+      contractWith('enum Color { Red, Green, Blue }'),
+      contractWith('constructor () BaseContract(1) {  }'),
+      'contract Token { string private name; }',
+      funcWith('assembly { let x := 1 }'),
+    ]
+
+    VALID_BLOCKS.forEach((curData) =>
+      it(`should not raise warn for valid block: ${label(curData)}`, () => {
+        const report = linter.processStr(curData, {
+          rules: { 'no-empty-blocks': 'warn' },
+        })
+        assertNoWarnings(report)
```

Comment:
since we already had the false negative caused by a syntax error, I'd say we add an assertion on the report not including any errors either (here and for all other no-warning testcases in this suite). We should also consider editing the `assertNoWarnings` function to check both errors and warnings, if it's not also thoroughly used to check no warnings are emitted in the presence of errors

Reply:
seems reasonable. Did you create the issue?